nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,799 @@
+
+        `screening_percentile_` : float
+            Screening percentile corrected according to volume of mask,
+            relative to the volume of standard brain.
+
+        `coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the mean of the models weight vector across
+            fold for each class.
+
+        `intercept_` : narray, shape (nclasses -1,)
+            Intercept (a.k.a. bias) added to the decision function.
+            It is available only when parameter intercept is set to True.
+
+        `cv_` : list of pairs of lists
+            List of the (n_folds,) folds. For the corresponding fold,
+            each pair is composed of two lists of indices,
+            one for the train samples and one for the test samples.
+
+        `std_coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the standard deviation of the models weight vector across
+            fold for each class.
+
+        `coef_img_`: dict of NiImage
+            Dictionary containing `coef_` with class names as keys,
+            and `coef_` transformed in NiImages as values. In the case
+            of a regression, it contains a single NiImage at the key 'beta'.
+
+        `std_coef_img_`: dict of NiImage
+            Dictionary containing `std_coef_` with class names as keys,
+            and `coef_` transformed in NiImages as values. In the case
+            of a regression, it contains a single NiImage at the key 'beta'.
+
+        `cv_y_true_` : numpy.ndarray, shape=(n_samples * n_folds, n_classes)
+            Ground truth labels for left out samples in inner cross-validation.
+
+        `cv_y_pred_` : numpy.ndarray, shape=(n_samples * n_folds, n_classes)
+            Predicted labels for left out samples in inner cross-validation.
+
+        `cv_params_`: dict of lists
+            Best point in the parameter grid for each tested fold
+            in the inner cross validation loop.
+
+        `cv_indices_` : numpy.ndarray, shape=(n_samples * n_folds)
+            Indices of the inner cross-validation folds.
+
+        `cv_scores_` : dict, (classes, n_folds)
+            Scores (misclassification) for each parameter, and on each fold
+        """
+        # Setup memory, parallel and masker
+        if isinstance(self.memory, _basestring) or (self.memory is None):
+            self.memory_ = Memory(cachedir=self.memory, verbose=self.verbose)
+        else:
+            self.memory_ = self.memory
+
+        # nifti masking
+        self.masker_ = check_embedded_nifti_masker(self, multi_subject=False)
+        X = self.masker_.fit_transform(X)
+        self.mask_img_ = self.masker_.mask_img_
+
+        X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=np.float,
+                         multi_output=True, y_numeric=True)
+
+        # Setup model
+        if not isinstance(self.estimator, _basestring):
+            warnings.warn('Use a custom estimator at your own risk.')
```

Comment:
May be check is it an estimator and raise a warning then ?
Regarding warning message: "Estimator is expected as string {0} but you provided your own custom estimator {1}."